src/main/java/com/codeborne/selenide/webdriver/ChromeDriverFactory.java

```diff
@@ -54,16 +58,60 @@ private ChromeOptions transferChromeOptionsFromSystemProperties(ChromeOptions cu
       if (key.startsWith(prefix)) {
         String capability = key.substring(prefix.length());
         String value = System.getProperties().getProperty(key);
-        if (capability.equals("args")) {
-          List<String> args = Arrays.asList(value.split(","));
-          currentChromeOptions.addArguments(args);
-        } else {
-          log.warning(capability + "is ignored." +
-                  "Only so-called arguments (chromeoptions.args=<values comma separated>) " +
-                  "are supported for the chromeoptions at the moment");
+        switch (capability) {
+          case "args": {
+            List<String> args = Arrays.asList(value.split(","));
+            currentChromeOptions.addArguments(args);
+            break;
+          }
+          case "prefs": {
+            Map<String, Object> prefs = new HashMap<>();
+            String[] allPrefs = value.split(",");
+            for (String pref: allPrefs) {
+              String[] keyValue = pref.split("=");
+              if (keyValue.length == 1) {
+                log.warning("Missing '=' sign while parsing <key=value> pairs from "
+                        + capability + ". Key '" + keyValue[0] + "' is ignored.");
+                continue;
+              }
+              else if (keyValue.length > 2) {
+                log.warning("More than one '=' sign while parsing <key=value> pairs from "
+                        + capability + ". Key '" + keyValue[0] + "' is ignored.");
+                continue;
+              }
+              prefs.put(keyValue[0], parseString(keyValue[1]));
+            }
+            currentChromeOptions.setExperimentalOption("prefs", prefs);
+            break;
+          }
+          default:
+            log.warning(capability + " is ignored." +
+                    "Only so-called arguments (chromeoptions.args=<values comma separated>) " +
+                    "and preferences (chromeoptions.prefs=<comma-separated dictionary of key=value> " +
+                    "are supported for the chromeoptions at the moment");
+            break;
         }
       }
     }
     return currentChromeOptions;
   }
+
+  private Object parseString(String value) {
```

Comment:
lol, I need 10 minutes to understand how it works.
So prefs maps has Int/Boolean/Strings as value and you map "true/false" to boolean, numbers to int, and the rest stays string. Could you try to comment this (if I am right) in this method? :) It will help future readers to understand the magic, it looked strange to me at first....

Reply:
@vinogradoff ok